### ISSUE TITLE
Set equality is unexpectedly ordered

### ISSUE THREAD

**Issue by rnewman** [mjackson/expect#93]
```
  it('Compares sets correctly.', (done) => {
    expect(new Set(['a', 'b'])).toEqual(new Set(['b', 'a']));
    done();
  });
```

```
      Error: Expected Set (2) {'a', 'b'} to equal Set (2) {'b', 'a'}
      + expected - actual

      -["a","b"]
      +["b","a"]
```

This is with 1.18.0. I expect set equality to be independent of order; the order of elements is an implementation detail.

**Comment by ljharb** [comment]
Your expectation is incorrect - `Set` in JavaScript has a defined and consistent implementation order, and is a spec detail, not an implementation detail.

**Comment by rnewman** [comment]
I wouldn't expect the preserved insertion order of the set — which is noted in the spec — to influence its equality properties, but hey, JS is full of surprises. Time to switch to `Immutable.js` for this, I guess.

**Comment by ljharb** [comment]
I think it'd be reasonable to have a method that explicitly checked that two Sets/Maps/arrays had the same contents, irrespective of order.

**Comment by mjackson** [comment]
@ljharb What would you call the `to` method?

**Comment by ljharb** [comment]
hmm - naming is hard :-) something like `expect(foo).toBeTheSameSetAs(bar)` would work but that uses the math definition of Set, not the JS one. `expect(foo).toMatch(bar)` maybe?

**Comment by mjackson** [comment]
How about `expect(a).toEqualSet(b)`? [expect-jsx](https://github.com/algolia/expect-jsx) already uses `toEqualJSX`.

**Comment by mjackson** [comment]
Actually, just found [this](http://www.solitaryroad.com/c725.html):

> Two sets A and B are said to be equal if every element of A is an element of B and every element of B is an element of A. The equality of sets A and B is denoted by A = B. Inequality of two sets A and B is denoted by A ≠ B.

Seems like we should just do some type detection in `toEqual` and `toNotEqual` and compare sets by their contents, irrespective of order.

**Comment by ljharb** [comment]
@mjackson if it's generic, that would break tests that currently compare array orderings. If it's not - and only for `Set`/`Map` - then you only have a slow `try/catch` way of brand-checking that it's a Set or a Map, *and*, Sets and Maps in JS *are* implicitly ordered.

**Comment by mjackson** [comment]
Even though they are implicitly ordered, conceptually a Set is not. toEqual
is the right method IMO.

**Comment by ljharb** [comment]
I don't agree in the case of JS - but it wouldn't be a tragedy imo to have `toEqual` compare Sets and Maps in an unordered fashion - if i want ordering, I can `[...foo]` and compare those.

**Comment by mjackson** [comment]
I'd be open to a PR that does this, @rnewman, if you're up for it. :)
